src/robot/running/runner.py

```diff
@@ -112,13 +112,13 @@ def visit_test(self, test):
             self._output.warn("Multiple test cases with name '%s' executed in "
                               "test suite '%s'." % (test.name, self._suite.longname))
         self._executed_tests[test.name] = True
-        result = self._suite.tests.create(name=test.name,
+        result = self._suite.tests.create(name=self._resolve_setting(test.name),
                                           doc=self._resolve_setting(test.doc),
                                           tags=self._resolve_setting(test.tags),
                                           starttime=get_timestamp(),
                                           timeout=self._get_timeout(test))
         self._context.start_test(result)
-        self._output.start_test(ModelCombiner(test, result))
+        self._output.start_test(ModelCombiner(test, result, unresolved_name=test.name))
```

Comment:
As I commented above, I don't think `unresolved_name` is a useful attribute to add to the listener interface. `resolved_name` could be, but I think it can be added later if really needed.

The way to give the original name to listener, without needing the `unresolved_name` later, is just using `name=test.name` here.

Reply:
I apply your proposition for ListenerArguments, I had misunderstood how works ListenerArguments. But now, the testcases name in the output.xml  is not resolved. 
It's due to `def start_test(self, test)` of `Logger` who take in argument the `ModelCombiner` class and when the method call `test.name`, it's a unresolved testcase name who is returned.
I don't kwow how we can have unresolved name in listenerv2 and the output.xml. Do you want really the unresolved name in listenerv2? Otherwise, i think we need to modify more code. What do you think?

Reply:
Good point that if we change the name used by ModelCombiner then also others who use it see that name. We have three possibilities what to do with listener v2:

1. Just use the resolved name. Easiest to implement but @michalanglart [commented](https://github.com/robotframework/robotframework/issues/2962#issuecomment-551978253) earlier that this would cause some problems for them.

2. Use the resolved name but also add `originalname` to the listener attributes. This would be easy to implement as well and would give @michalanglart an easy way to handle this situation in their code. Would also have the benefit of showing both names.

3. Try to handle this special case in the listener side. ModelCombiner has access to both the original and the resolved names so this ought to be doable. Name is handled differently than other attributes, so this would be a bit more work than option 2.

Option 2. starts to sound best for me. On the listener side it shouldn't require more changes than this:

```diff
--- a/src/robot/output/listenerarguments.py
+++ b/src/robot/output/listenerarguments.py
@@ -117,7 +117,8 @@ class StartTestArguments(_ListenerArgumentsFromItem):
 
     def _get_extra_attributes(self, test):
         return {'critical': 'yes' if test.critical else 'no',
-                'template': test.template or ''}
+                'template': test.template or '',
+                'originalname': test.data.name}
```

Notice that I used `originalname`, not `original_name` on purpose for consistency reasons. We've used that style with other attributes as well (which may not have been a great idea but cannot really be changed).

Reply:
I applied the modification, i updated test and the documentation for listeners v2.